### ISSUE TITLE
Skipping Front matter of Jekyll

### ISSUE THREAD

**Issue by rugk** [tonyganch/gonzales-pe#195]
From https://github.com/sasstools/sass-lint/issues/542:

---

Some static site generater such as Jekyll include the [YAML front matter block](http://jekyllrb.com/docs/frontmatter/) at the top of the scss file for their own tracking purposes. [Front matter is necessary to compile.](http://jekyllrb.com/docs/assets/)

``` css

---
layout: null
someProp: value

---


html {
    background-color: #fff;
}
```

However, syntax error occurs in the sass-lint because the line of triple dashes `---`.
Here is sass-lint's error log (I'm using SublimeLinter-contrib-sass-lint). And linter does not work anymore.

```
c:/some/path/style.scss
  1:1  error  Please check validity of the block starting from line #1  Fatal

✖ 1 problem (1 error, 0 warnings) 
```

Can anyone help how to solve this issue?
How about add an option for skipping front matter. see [haml-lint#skipping-frontmatter](https://github.com/brigade/haml-lint#skipping-frontmatter)

See also:
- [Setting up sass with Jekyll](http://blog.davepoon.net/2015/01/19/setting-up-sass-with-jekyll/)

**Comment by tonyganch** [comment]
cc @DanPurdy 

GPE is just a parser, it assumes that the text that you pass is sass, but what you actually give is yaml.
The solution should be outside the parser: cut yaml part somehow and pass the rest (actual sass part) to the parser.
Sorry, this is a wontfix from our side.

**Comment by rugk** [comment]
Mhh...
So do you think this "cut-away" could be included in sass-lint?

**Comment by tonyganch** [comment]
@DanPurdy @bgriffith, guys what do you think?

**Comment by rugk** [comment]
Especially /cc @sasstools @woneob...

**Comment by DanPurdy** [comment]
It's a very grey area I think. On one hand sass-lint doesn't actually care about what files you pass it, only the result we get from the AST but as @tonyganch rightfully says gonzales does care and should only be fed Sass. 

The problem for me is that sass-lint would have to read a file interpret it and act on it all before passing it to the AST but im probably leaning more to that than adding support in gonzales on second thoughts.

Basically, i don't know the best way forward right now.
